hooks/tk-multi-publish2/basic/publish_session_geometry.py

```diff
@@ -0,0 +1,514 @@
+
+    def accept(self, settings, item):
+        """
+        Method called by the publisher to determine if an item is of any
+        interest to this plugin. Only items matching the filters defined via the
+        item_filters property will be presented to this method.
+
+        A publish task will be generated for each item accepted here. Returns a
+        dictionary with the following booleans:
+
+            - accepted: Indicates if the plugin is interested in this value at
+                all. Required.
+            - enabled: If True, the plugin will be enabled in the UI, otherwise
+                it will be disabled. Optional, True by default.
+            - visible: If True, the plugin will be visible in the UI, otherwise
+                it will be hidden. Optional, True by default.
+            - checked: If True, the plugin will be checked in the UI, otherwise
+                it will be unchecked. Optional, True by default.
+
+        :param settings: Dictionary of Settings. The keys are strings, matching
+            the keys returned in the settings property. The values are `Setting`
+            instances.
+        :param item: Item to process
+
+        :returns: dictionary with boolean keys accepted, required and enabled
+        """
+
+        accepted = True
+        publish_template = self._get_template("Publish file Template", settings)
+        if not publish_template and not item.parent.properties["work_file_template"]:
+            accepted = False
+
+        # check that the AbcExport command is available!
+        if not mel.eval("exists \"AbcExport\""):
+            accepted = False
+            self.logger.warn("Item not accepted because export command AbcExport is not available.")
+        
+        return {
+            "accepted": accepted,
+            "checked": True
+        }
+
+    def validate(self, settings, item):
+        """
+        Validates the given item to check that it is ok to publish. Returns a
+        boolean to indicate validity.
+
+        :param settings: Dictionary of Settings. The keys are strings, matching
+            the keys returned in the settings property. The values are `Setting`
+            instances.
+        :param item: Item to process
+        :returns: True if item is valid, False otherwise.
+        """
+
+        # check that there is still geometry in the scene:
+        if not cmds.ls(geometry=True, noIntermediate=True):
+            self.logger.warn("Validation failed because there is no geometry in the scene to be exported")
+            return False
+
+        publisher = self.parent
+
+        # ensure we have an updated project root
+        project_root = cmds.workspace(q=True, rootDirectory=True)
+        item.properties["project_root"] = project_root
+
+        # get the configured work file template
+        work_template = item.parent.properties["work_file_template"]
+        publish_template = self._get_template("Publish file Template", settings)
```

Comment:
you could get the publish template once in `accept` and store it on `item` so that you don't have to get it again here. Would also allow you to just move the logic in `_get_template()` up into `accept`.

Reply:
Reworked. Removed _get_template() and got code up